### ISSUE TITLE
[Feature] Upload to F-Droid and releases in GitHub 

### ISSUE THREAD

**Issue by Wladefant** [TUM-Dev/Campus-Flutter#89]
All open Source apps must be uploaded to fdroid. You should also create automated  github releases from which all teh other app stores can fetch the new release. Easy to set up and is perfect. Pre releases and so on will be easy to create

**Comment by jakobkoerber** [comment]
Thank you for your suggestions! Do I understand you correctly that uploading open source apps to F-Droid is mandatory? And which other App Stores do you mean?

**Comment by CommanderStorm** [comment]
No, uploading apps to alternative app stores like [F-Droid](https://f-droid.org/en/), [Aurora](https://gitlab.com/AuroraOSS/AuroraStore) or similar is not mandatory. 

In fact, this is the other way around (source: https://f-droid.org/docs/Inclusion_Policy/):
All Apps **on** said store have to be 100% open source (exceptions being made for images/movies, but not for build / other tooling).

This imposes heavy restrictions, for example:
- this prohibits the usage of `crashytics`, `sentry` or `exgen` to monitor errors
- prohibits the usage of google play services to get a more acurate wifi signal or other functionality which relies on these services
  For our App, this might be problematic for notifications and TUM-Keyless (released soon ™️, depending on the Google/Apple Wallet)  as I understand this.
- the usage of `beta`s  (German: ["irgendwelcher Toolchains für Vorabversionen"](https://f-droid.org/docs/Inclusion_Policy))
  ⇒ @Wladefant "Pre releases and so on will be easy to create" is not true as far as I read their documentation


In fact, the old campus app was not on said store for [multiple reasons](https://github.com/TUM-Dev/Campus-Android/issues?q=is%3Aissue+f-droid+is%3Aclosed). Most notably the usage of
- google cloud messaging for chatting and notifications (now partially removed)
- crashlytics for monitoring crashes
- more acurate location services indoors

See https://github.com/TUM-Dev/Campus-Android/issues/226 for further details.
Core-maintainers doing this work is not feasable, given the amount of users

I'd suggest the previous deal:
If the F-Droid crowd provides the PRs to get this functionality, and said functionality it is not significantly more effort/harder to maintain because of that, we will accept this.

**Comment by jakobkoerber** [comment]
Thanks for that extensive explanation! 👌🏼 If I understand you correctly that would mean not being able to use Firebase Cloud Messaging and the Google Maps SDK etc.? If so, I don't see the advantage of "crippling" our app and preventing potential features just to have it on some store and would therefore suggest to close this issue.

**Comment by Wladefant** [comment]
@jakobkoerber sorry for the misunderstanding I said "mandatory" jokingly. Could create a PR in the feature.

@CommanderStorm when I was talking about the pre releases I was talking about github releases. I talked about the management of the app stores through github with the help of actions. No need to manually upload each release

**Comment by CommanderStorm** [comment]
@jakobkoerber I would not be so harsh as to say crippling.

Most issues can be worked around, but require someone (not Jakob/me) to look into.
If @Wladefant or somebody else wants to tackle this, that would be appreciated (as by above offer ^^).

- Cloud Messaging is not nessesary for this app to work (we don't have a chat). If users get clearly communicated that they won't get `alarm` or other notifications, this is fine.
- Google Maps SDK can be replaced by other SDKs like [flutter-maplibre-gl](https://github.com/maplibre/flutter-maplibre-gl) with the [Tileserver of NavigaTUM](https://nav.tum.de/maps/)

**Comment by jakobkoerber** [comment]
> I talked about the management of the app stores through github with the help of actions. No need to manually upload each release

This is already in place for the Apple App Store, Google Play Store and the deployment of the web application

**Comment by jakobkoerber** [comment]
> @jakobkoerber I would not be so harsh as to say crippling.

Yes, crippling might be exaggerated, but my point stands as I don't see the benefit of killing potential features that might be relevant in the feature to have the application on an app store that is used by a very small percentage of users. Could this maybe also be the reason why Campus-Android is not on F-Droid?

> * Cloud Messaging is not nessesary for this app to work (we don't have a chat). If users get clearly communicated that they won't get `alarm` or other notifications, this is fine.

Firebase Cloud Messaging is as far as I can tell the easiest way to enable cross-platform notifications. Is your proposal to disable notifications for all users or just for the F-Droid etc. users?

**Comment by philippzagar** [comment]
So, to jump in for a second:
@Wladefant Thanks a lot for raising the issue, it's always nice to attract new students interested in contributing to the TUM Devs 😃 

Moving to a "fully open-source" application is definitly out of scope at the current stage of the application and development progress. We definitly will use Firebase, Maps SDK etc. in the TCA Flutter application, as it simplifies the implementation quite a bit (Flutter was designed to be that way!). 
As we are all students who volunteer their time to develop this application, we should always choose the path that benefits the majority of our userbase, not tiny percentages.

@Wladefant If you have spare resources, feel free to create a version which complies with the "fully open source" requirements from F-Droid. However, be prepared to go the entire way and also maintain the code.

**Comment by Wladefant** [comment]
@philippzagar yes fresh student here. Haven't really looked in teh code base or anything. Just throwing in some suggestions. Haven't really worked with flutter either.

**Comment by CommanderStorm** [comment]
I am going to close this issue as I think the consensus is clear:

If there is support from the FDroid crowd, we will make the app available on alternative stores. Said support includes both the nessesary PRs and help in the maintenance effort created by this shift
At the current state, the app would be rejected outright.

However: One critical requirement is, that this would not impact the other stores or developer productivity negatively.
As FDroid does not allow [`beta`](https://github.com/TUM-Dev/Campus-Flutter/issues/89#issuecomment-1722340176)s or [proprietary software like crashlytics](https://github.com/TUM-Dev/Campus-Flutter/issues/89#issuecomment-1722340176), this might be harder than the above requirement.

As it stands, the “core team” (we are just a tiny number of people doing this in our spare time) will not work on this as the cost/benefits are not feasible for such a small team.
We can reopen, if something significant changes in the landscape (i.e. more than a small percentile use said store), or if @Wladefant or other FDroid users help us to get there. ^^

**Comment by kordianbruck** [comment]
Just for some context: we had the Campus App in FDroid for a very long time, until F-Droid decided themselves to kick out a lot of apps that didn't "fully do opensource". So while at some point in the past it was reasonable amount of work to maintain it in parallel, F-Droid has basically lost us here on their own accord. I agree with everyone else: its too much work and time not spend on actual more important work.
